react-native-purchases-ui/ios/RCPaywallFooterViewManager.m

```diff
@@ -0,0 +1,144 @@
+//
+//  RCPaywallFooterViewManager.m
+//  RNPaywalls
+//
+//  Created by Cesar de la Vega on 29/12/23.
+//
+
+#import "RNPaywalls.h"
+#import "RCPaywallFooterViewManager.h"
+
+@import RevenueCatUI;
+@import PurchasesHybridCommon;
+
+#import "UIView+Extensions.h"
+
+#import <React/RCTShadowView.h>
+#import <React/RCTUIManager.h>
+#import <React/RCTBridge.h>
+#import <React/RCTRootViewDelegate.h>
+#import <React/RCTEventEmitter.h>
+
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface FooterViewWrapper : UIView
+
+- (instancetype)initWithFooterViewController:(UIViewController *)footerViewController 
+                                      bridge:(RCTBridge *)bridge;
+
+@end
+
+NS_ASSUME_NONNULL_END
+
+@interface FooterViewWrapper () <RCPaywallViewControllerDelegate>
+
+@property (strong, nonatomic) UIViewController *footerViewController;
+@property (strong, nonatomic) RCTBridge *bridge;
+
+@property BOOL addedToHierarchy;
+
+@end
+
+@implementation FooterViewWrapper
+
+- (instancetype)initWithFooterViewController:(UIViewController *)footerViewController bridge:(RCTBridge *)bridge {
+    if ((self = [super initWithFrame:footerViewController.view.bounds])) {
+        _bridge = bridge;
+        _footerViewController = footerViewController;
+    }
+
+    return self;
+}
+
+- (void)safeAreaInsetsDidChange {
+    [super safeAreaInsetsDidChange];
+
+    // Get the safe area insets, for example
+    UIEdgeInsets safeAreaInsets = self.safeAreaInsets;
+
+//    TODO: figure out a better way of sending event, since this is deprecated
+//    It's probably better to create a singleton from the module that this view manager can call and use to send events
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+    [self.bridge.eventDispatcher sendAppEventWithName:safeAreaInsetsDidChangeEvent
+                                                 body:@{@"top": @(safeAreaInsets.top),
+                                                        @"left": @(safeAreaInsets.left),
+                                                        @"bottom": @(safeAreaInsets.bottom),
+                                                        @"right": @(safeAreaInsets.right)}];
+#pragma clang diagnostic pop
+}
+
+- (void)paywallViewController:(RCPaywallViewController *)controller didChangeSizeTo:(CGSize)size API_AVAILABLE(ios(15.0)){
+    [_bridge.uiManager setSize:size forView:self];
+}
+
+- (void)layoutSubviews {
+    [super layoutSubviews];
+
+    if (!self.addedToHierarchy) {
+        UIViewController *parentController = self.parentViewController;
+        if (parentController) {
+            self.footerViewController.view.translatesAutoresizingMaskIntoConstraints = NO;
```

Comment:
definitely merits a comment like the one on Android side

Reply:
👍🏻